cdf_fabric_replicator/extractor.py

```diff
@@ -304,19 +315,52 @@ def get_events(self, df: DataFrame) -> list[EventWrite]:
             events.append(new_event)
         return events
 
-    def convert_lakehouse_data_to_df(self, file_path: str, token: str) -> DataFrame:
+    def retrieve_external_ids_from_lakehouse(
+        self, file_path: str, token: str
+    ) -> list[str]:
         try:
-            dt = DeltaTable(
-                file_path,
-                storage_options={"bearer_token": token, "user_fabric_endpoint": "true"},
+            dt = DeltaTable(file_path, storage_options={"bearer_token": token})
+            table = dt.to_pyarrow_table(columns=["externalId"])
+            return pc.unique(table.column("externalId")).to_pylist()
+
+        except Exception as e:
+            self.logger.error(
+                f"Error while converting lakehouse data to DataFrame: {e}"
             )
+            raise e
+
+    def convert_lakehouse_data_to_df(self, file_path: str, token: str) -> DataFrame:
+        try:
+            dt = DeltaTable(file_path, storage_options={"bearer_token": token})
             return dt.to_pandas()
         except Exception as e:
             self.logger.error(
                 f"Error while converting lakehouse data to DataFrame: {e}"
             )
             raise e
 
+    def convert_lakehouse_data_to_df_batch(
+        self, file_path: str, external_id: str, timestamp: str, token: str
+    ) -> Generator[DataFrame, None, None]:
+        try:
+            dt = DeltaTable(file_path, storage_options={"bearer_token": token})
```

Comment:
Will this do the actual fetch of filtered data happen when the filter is applied is applied down at line 352?